docs/RFCS/freeze.md

```diff
@@ -0,0 +1,102 @@
+- Feature Name: Data/network freeze
+- Status: draft
+- Start Date: 2016-02-18
+- Authors: Ben Darnell
+- RFC PR: #4499
+- Cockroach Issue:
+
+
+# Summary
+
+This RFC outlines the plan for freezing our data formats and network
+protocols.
+
+# Motivation
+
+We currently make backwards-incompatible changes to data formats
+without providing any means to upgrade without data loss. This will
+need to stop before beta for obvious reasons.
+
+# Detailed design
+
+## Freeze plan
+
+The freeze will proceed in several steps.
+
+### Stage 0 (now)
+
+Anything goes; changes to on-disk formats do not require any kind of
+migration path.
+
+### Stage 1: Guaranteed upgrade path (Mar 24, 2016)
+
+In stage 1, we require backwards-compatibility with data written by
+any previous stage 1 build. It should always be possible to upgrade by
+stopping all of the old nodes and then bringing up the new version.
+It's OK at this stage if old and new versions cannot be run
+concurrently, or if the migration process takes some time (e.g.
+rewriting all data before the node can start up).
+
+It is acceptable at this stage if the process is somewhat manual (e.g.
+stop all the nodes, run `cockroach migrate` on each one, then restart
```

Comment:
I agree with the acceptability of a manual process. Rather than a per-node `migrate` command, I think we should initially target a `dump/restore` process that operates at the level of SQL metadata (tables, indexes, permissions, etc) and data (rows). I think this approach would handle more cases (a per-node `migrate` seems like it would have difficulty with a SQL data format change) and has a more straightforward story for downgrades. It definitely isn't as operator friendly, but I bet we'll want the flexibility.